p2p/host/relaysvc/relay_test.go

```diff
@@ -0,0 +1,67 @@
+package relaysvc
+
+import (
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/event"
+	"github.com/libp2p/go-libp2p/core/network"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/blank"
+	"github.com/libp2p/go-libp2p/p2p/host/eventbus"
+	swarmt "github.com/libp2p/go-libp2p/p2p/net/swarm/testing"
+	relayv2 "github.com/libp2p/go-libp2p/p2p/protocol/circuitv2/relay"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReachabilityChangeEvent(t *testing.T) {
+	h := bhost.NewBlankHost(swarmt.GenSwarm(t))
+	rmgr := NewRelayManager(h)
+	emitter, err := rmgr.host.EventBus().Emitter(new(event.EvtLocalReachabilityChanged), eventbus.Stateful)
+	if err != nil {
+		t.Fatal(err)
+	}
+	evt := event.EvtLocalReachabilityChanged{Reachability: network.ReachabilityPublic}
+	emitter.Emit(evt)
+	require.Eventually(
+		t,
+		func() bool { rmgr.mutex.Lock(); defer rmgr.mutex.Unlock(); return rmgr.relay != nil },
+		200*time.Millisecond,
```

Comment:
Better to increase this to a second, CI is super slow (sometimes), which then leads to flaky tests.

Reply:
fixed.